manifests/component/lookup_table.pp

```diff
@@ -0,0 +1,39 @@
+define rsyslog::component::lookup_table (
+  Integer          $priority,
+  String           $target,
+  String           $confdir,
+  Hash             $lookup_json,
+  String           $lookup_file,
+  Boolean          $reload_on_hup,
+  Optional[String] $format = '<%= $content %>'
+) {
+
+  include rsyslog
+
+  file { "rsyslog::component::lookup_table_json::${title}":
+    path    => "${target}.json",
+    content => inline_template('<%= JSON.pretty_generate @lookup_json %>'),
```

Comment:
In all my tests it does return the same string with keys in the same order. I'm actually testing for this in the unit tests by reading the `spec/fixtures/test_files/example_lookup.json` file and compare the contents of that file with the generated content from the code.